tools/halide_image_io.h

```diff
@@ -338,484 +399,342 @@ bool load_png(const std::string &filename, ImageType *im) {
 
     png_read_info(png_ptr, info_ptr);
 
-    int width = png_get_image_width(png_ptr, info_ptr);
-    int height = png_get_image_height(png_ptr, info_ptr);
-    int channels = png_get_channels(png_ptr, info_ptr);
-    int bit_depth = png_get_bit_depth(png_ptr, info_ptr);
-
-    // Expand low-bpp images to have only 1 pixel per byte (As opposed to tight packing)
-    if (bit_depth < 8) {
-        png_set_packing(png_ptr);
-    }
+    const int width = png_get_image_width(png_ptr, info_ptr);
+    const int height = png_get_image_height(png_ptr, info_ptr);
+    const int channels = png_get_channels(png_ptr, info_ptr);
+    const int bit_depth = png_get_bit_depth(png_ptr, info_ptr);
 
+    const halide_type_t im_type(halide_type_uint, bit_depth);
+    std::vector<int> im_dimensions = { width, height };
     if (channels != 1) {
-        *im = ImageType(width, height, channels);
-    } else {
-        *im = ImageType(width, height);
+        im_dimensions.push_back(channels);
     }
 
-    png_set_interlace_handling(png_ptr);
```

Comment:
Yes, because AFAICT it's a no-op here: the docs indicate it's only used when writing an image, not when reading it.